src/librustc_data_structures/sharded.rs

```diff
@@ -69,6 +70,8 @@ impl<T> Sharded<T> {
         }
     }
 
+    /// Get a shard with a pre-computed hash value, which must be computed by
+    /// `FxHasher`.
```

Comment:
It doesn't matter which hash you use here as long as you use the same one for a single instance of `Sharded`.